# use the names variable to store your list of file names
SAMPLES = ["P0rep1", "P0rep2", "P4rep1", "P4rep2", "P7rep1", "P7rep2", "ADrep1", "ADrep2"] 
READS = ["R1", "R2"]

# your snakemake rule should produce the output from MultiQC and if you link your rules inputs
# and outputs, you should only have one file listed in your rule all
rule all:
    input:
        "/projectnb/bf528/students/neharao/project_1_rnaseq/results/multiqc_report.html"

# make sure to read the fastqc manual to get a sense for appropriate running command
# we have given you the basic structure of the snakemake rule to get you used to how they
# are structured
rule fastqc:
    input:
        "/projectnb/bf528/students/neharao/project_1_rnaseq/subsampled_files/{sample}subsample_{read}.fastq.gz"
    output:
        "/projectnb/bf528/students/neharao/project_1_rnaseq/results/{sample}subsample_{read}_fastqc.html"
    shell:
        "fastqc {input} -o /projectnb/bf528/students/neharao/project_1_rnaseq/results/"

# remember that you want multiqc to run only after fastqc has run on all the files
rule multiqc:
    input:
        expand("/projectnb/bf528/students/neharao/project_1_rnaseq/results/{sample}subsample_{read}_fastqc.html", sample=SAMPLES, read=READS)
    output:
        "/projectnb/bf528/students/neharao/project_1_rnaseq/results/multiqc_report.html"
    shell:
        "multiqc -o /projectnb/bf528/students/neharao/project_1_rnaseq/results/ /projectnb/bf528/students/neharao/project_1_rnaseq/subsampled_files/"
